Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -203,146 +203,87 @@ function xASL_wrp_ResampleASL(x)
-            PWI = ASL_im;
-        end
-
-        % Save PWI4D
-        fprintf('%s\n', PathPWI4D{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, PWI, 32, false);
-        
-        % Create single PWI for further steps in ASL module
-        PWIsingle = xASL_stat_MeanNan(PWI(:,:,:,1:x.Q.NumberEchoTimes:end),4); % Average across PLDs from each first TE
-        
-        % Save single PWI
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWIsingle, 32, false);
-        
-    else
-        % Paired subtraction
-        [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
-        ASL_im = ControlIm - LabelIm;
-
-        % Average PWI - multiPLD
-        if x.modules.asl.bMultiPLD 
-			% Skip every other value in x.Q.Initial_PLD as it was stored for both control and label images 
-			% and we need the PLD vector now for the pairwise subtractions only
-			Initial_PLD_PWI = x.Q.Initial_PLD(1:2:end);
-
-			% The labeling durations also need to be taken into account and unique combinations of PLDs with LDs should be considered
-			if numel(x.Q.Initial_PLD) == numel(x.Q.LabelingDuration)
-				Initial_LabDur_PWI = x.Q.LabelingDuration(1:2:end);
-			else
-				Initial_LabDur_PWI = ones(size(Initial_PLD_PWI)) * x.Q.LabelingDuration;
-			end
-        
-			% After averaging across PLDs, we'll obtain these unique PLDs+LD combinations
-			% indexNew lists for each original position to where it should be averaged
-			[~, ~, indexNew] = unique([Initial_PLD_PWI, Initial_LabDur_PWI], 'stable', 'rows');
-			
-            % MultiPLD PWI after averaging
-			PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), max(indexNew)); % preallocate PWI
-            for nPLD = 1:max(indexNew)
-                PWI(:, :, :, nPLD) = xASL_stat_MeanNan(ASL_im(:, :, :, indexNew == nPLD), 4); % Averaged PWI4D 
-            end
-            
-            % Save PWI
-            fprintf('%s\n', PathPWI4D{iSpace});
-            xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, PWI, 32, false);
-            
-            PWIsingle = xASL_stat_MeanNan(ASL_im, 4); % create single PWI for further steps in ASL module
-            
-            % Save single PWI
-            fprintf('%s\n', PathPWI{iSpace});
-            xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWIsingle, 32, false);
-            
-        else
-            PWI = xASL_stat_MeanNan(ASL_im, 4); % singlePLD PWI
-            % Save PWI
-            fprintf('%s\n', PathPWI{iSpace});
-            xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWI, 32, false);
-        end            
+    % Here we reload this again, because we have different 
+    ASL4D = xASL_io_Nifti2Im(PathASL4D{iSpace}); % Load time-series nifti
+    
+    if numel(size(ASL4D))>4
+        warning('In BIDS ASL NIfTIs should have [X Y Z n/PLD/TE/etc] as 4 dimensions');

```

Comment:
Done in 44652fe6cc406d1d7a04b431c5911ccb51db33d0. The warning had a slightly different message to explain the error, so now converted to fprintf instead of removed.

Reply:
Perfect.